server/src/main/java/com/octopus/teamcity/opentelemetry/server/TeamCityBuildListener.java

```diff
@@ -113,21 +53,21 @@ private String mask(String message) {
 
     @Override
     public void buildStarted(@NotNull SRunningBuild build) {
-        if (pluginReady()) {
+        if (buildListenerReady()) {
             String buildId = getBuildId(build);
             String buildName = getBuildName(build);
             Loggers.SERVER.debug("OTEL_PLUGIN: Build started method triggered for " + buildName);
 
-            Span parentSpan = getParentSpan(build);
-            Span span = createSpan(buildId, parentSpan);
-            this.spanMap.put(buildId, span);
-            Loggers.SERVER.info("OTEL_PLUGIN: Tracer initialized and span created for " + buildName);
+            String parentBuildId = getParentBuild(build);
+            Span parentSpan = this.otelHelper.getParentSpan(parentBuildId);
+            Span span = this.otelHelper.createSpan(buildId, parentSpan);
+            this.otelHelper.addSpanToMap(buildId, span);
```

Comment:
Should we combined these, so that it auto-adds to the span?

Reply:
Yep these have been combined and updated